docs/docs/DesignPrinciples/index.md

```diff
@@ -0,0 +1,34 @@
+# Design Principles
```

Comment:
This is not really self-explanatory for an external doc page, especially if it is a "top-level" item, maybe at least put it under the "implementation" section.

Reply:
I moved the design principles to the implementation subpages as suggested. I would not extend that declaration further but I am open to suggestions